contracts/Strategy.sol

```diff
@@ -784,10 +768,33 @@ contract Strategy is BaseStrategy, DydxFlashloanBase, ICallee {
         Account.Info memory account,
         bytes memory data
     ) public override {
-        (bool deficit, uint256 amount, uint256 repayAmount) = abi.decode(data, (bool, uint256, uint256));
+        (bool deficit, uint256 amount) = abi.decode(data, (bool, uint256));
         require(msg.sender == SOLO);
+        require(sender == address(this));
+        uint256 repayAmount = amount.add(2);
+
+        uint256 bal = want.balanceOf(address(this));
+        require(bal >= amount); // to stop malicious calls
+
+        IWETH(weth).withdraw(bal);
+
+        //if in deficit we repay amount and then withdraw
+        if (deficit) {
+
+            cToken.repayBorrow{value: amount}();
+
+            //if we are withdrawing we take more to cover fee
+            cToken.redeemUnderlying(repayAmount);
```

Comment:
should we add require == 0?

Reply:
I think this one is safe? worse case we repay some borrow but dont redeem

Reply:
ahh you are right. will move

Reply:
fe028c8eea2e426bf9df7e11c8035027430abf76